File: ui/react/src/action/UpdateForm.ts
import SheetAction from "./SheetAction";
import Form from "../model/Form";
import Action from "./Action";

class UpdateForm extends SheetAction {
    public form: Form;

    public static deserialize(input: any): UpdateForm {
        const sheetName = input.sheetName;
        const form = Form.deserialize(input.form);
        return new UpdateForm(sheetName, form);
    }

    constructor(sheetName: string, form: Form) {
        super(sheetName);
        this.form = form;
    }

    public isAssetAction(): boolean {
        return true;
    }

    public targetAsset(): string | undefined {
        return this.form.name;
    }

    public wrapper(): Action {
        const action = new Action();
        action.updateForm = this;
        return action;
    }
}

export default UpdateForm;

File: core/src/main/java/com/ctrip/ferriswheel/core/action/UpdateForm.java
/*
 * MIT License
 *
 * Copyright (c) 2018-2019 Ctrip.com
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.common.action.Action;
import com.ctrip.ferriswheel.common.form.Form;

public class UpdateForm extends SheetAction implements Action {
    private String formName;
    private Form formData;

    public UpdateForm() {
    }

    public UpdateForm(String sheetName, String formName, Form formData) {
        super(sheetName);
        this.formName = formName;
        this.formData = formData;
    }

    public String getFormName() {
        return formName;
    }

    public void setFormName(String formName) {
        this.formName = formName;
    }

    public Form getFormData() {
        return formData;
    }

    public void setFormData(Form formData) {
        this.formData = formData;
    }
}
